From stable-bounces@linux.kernel.org Fri Mar  2 00:46:56 2007
From: Tejun Heo <htejun@gmail.com>
Date: Fri, 2 Mar 2007 17:45:30 +0900
Subject: libata: add missing PM callbacks
To: Guennadi Liakhovetski <g.liakhovetski@gmx.de>
Cc: <jgarzik@pobox.com>, stable@kernel.org, Alan <alan@lxorguk.ukuu.org.uk>
Message-ID: <20070302084530.GN20322@htj.dyndns.org>
Content-Disposition: inline

From: Tejun Heo <htejun@gmail.com>

Some LLDs were missing scsi device PM callbacks while having host/port
suspend support.  Add missing ones.

Signed-off-by: Tejun Heo <htejun@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/ata/pata_jmicron.c |    4 ++++
 drivers/ata/pata_sil680.c  |    4 ++++
 2 files changed, 8 insertions(+)

--- linux-2.6.20.1.orig/drivers/ata/pata_jmicron.c
+++ linux-2.6.20.1/drivers/ata/pata_jmicron.c
@@ -137,6 +137,10 @@ static struct scsi_host_template jmicron
 	.slave_destroy		= ata_scsi_slave_destroy,
 	/* Use standard CHS mapping rules */
 	.bios_param		= ata_std_bios_param,
+#ifdef CONFIG_PM
+	.suspend		= ata_scsi_device_suspend,
+	.resume			= ata_scsi_device_resume,
+#endif
 };
 
 static const struct ata_port_operations jmicron_ops = {
--- linux-2.6.20.1.orig/drivers/ata/pata_sil680.c
+++ linux-2.6.20.1/drivers/ata/pata_sil680.c
@@ -226,6 +226,10 @@ static struct scsi_host_template sil680_
 	.slave_configure	= ata_scsi_slave_config,
 	.slave_destroy		= ata_scsi_slave_destroy,
 	.bios_param		= ata_std_bios_param,
+#ifdef CONFIG_PM
+	.suspend		= ata_scsi_device_suspend,
+	.resume			= ata_scsi_device_resume,
+#endif
 };
 
 static struct ata_port_operations sil680_port_ops = {
